src/components/common/ErrorMessage.js

```diff
@@ -14,64 +14,101 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import React from 'react';
+import * as React from 'react';
+
+import type { ApiError } from '../../types/api-error';
 
 import './ErrorMessage.css';
 
 type ErrorMessageProps = {
   className?: string,
-  error:
-    | string
-    | {
-        message: string,
-        httpStatus?: any,
-        httpStatusText?: string,
-        httpUrl?: string,
-        httpQuery?: string,
-        httpBody?: string,
-      },
+  error: ApiError,
+};
+
+type SubPartProps = {
+  className?: string,
+  error: ApiError,
+  wrap?: boolean,
 };
 
 function ErrorAttr({ name, value }: { name: string, value: any }) {
   return (
-    <tr>
+    <tr className="ErrorMessage--detailItem">
       <td className="ErrorMessage--attr">{name}</td>
       <td className="ErrorMessage--value">{value}</td>
     </tr>
   );
 }
 
+function Message(props: SubPartProps) {
+  const { className, error, wrap } = props;
```

Comment:
This seems like we can simplify. What is the point of wrap?

```js
function Message({ error, className }) {
  const errorMsg = typeof error === 'string' ? error : error.message;
  return (
    <h3 className={`ErrorMessage--msg ${className || ''}`}>{errorMsg}</h3>
  );
}
```

Reply:
`wrap` is so the component can be split between the message and the details but still have consistent layout behavior.

In the notification, the message is used as the notification title and the details are the notification body. `wrap` keeps the layout the same whether the parts are split out and used individually like that or not.